src/torrent.ts

```diff
@@ -219,22 +227,89 @@ export async function loadTorrentDirLight(
 	return searchees;
 }
 
-export async function getTorrentByFuzzyName(
+function getKeysFromName(name: string): {
+	keyTitles: string[];
+	element?: string | number;
+	useFallback: boolean;
+} {
+	const stem = stripExtension(name);
+	const episodeAndKey = getEpisodeAndKey(stem);
+	if (episodeAndKey) {
+		const keyTitles = [episodeAndKey.keyTitle];
+		const element = `${episodeAndKey.season}.${episodeAndKey.episode}`;
+		return { keyTitles, element, useFallback: false };
+	}
+	const seasonKey = getSeasonKey(stem);
+	if (seasonKey) {
+		const keyTitles = [seasonKey.keyTitle];
+		const element = seasonKey.season;
+		return { keyTitles, element, useFallback: false };
+	}
+	const movieKey = getMovieKey(stem);
+	if (movieKey) {
+		const keyTitles = [movieKey.keyTitle];
+		return { keyTitles, useFallback: false };
+	}
+	const releaseAndKeys = getReleaseAndKeys(stem);
+	if (releaseAndKeys) {
+		const keyTitles = releaseAndKeys.keyTitles;
+		const element = releaseAndKeys.release;
+		return { keyTitles, element, useFallback: true };
+	}
+	return { keyTitles: [], useFallback: true };
+}
+
+export async function getTorrentByName(
 	name: string,
-): Promise<null | Metafile> {
+): Promise<{ keys: string[]; metas: Metafile[] }> {
+	const { keyTitles, element, useFallback } = getKeysFromName(name);
+	const metas = useFallback ? await getTorrentByFuzzyName(name) : [];
```

Comment:
how valuable is the fallback? it seems like this logic is pretty comprehensive. I'd be in favor of deleting the fuse code.

Reply:
The `ANIME_REGEX` cannot be as complete as the others due to anime naming. So we always use Fuse additionally.

Fuse is used exclusively if the media doesn't match one of our regexes. This is to support non-video matching from rss/announce.